geode-core/src/test/java/org/apache/geode/internal/cache/TXStateTest.java

```diff
@@ -390,4 +391,31 @@ public void firePendingCallbacksSetsChangeAppliedToCacheInEventLocalFilterInfo()
     verify(filterInfo1).setChangeAppliedToCache(true);
     verify(filterInfo2).setChangeAppliedToCache(true);
   }
+
+  @Test
+  public void commitConvertsCommitConflictExceptionToTransactionDataRebalancedExceptionIfCausedByRegionDestroyedExceptionFromReserveAndCheck() {
+    TXState txState = spy(new TXState(txStateProxy, false, disabledClock()));
+    CommitConflictException cce = new CommitConflictException("Conflict caused by cache exception");
+    RegionDestroyedException rde =
+        new RegionDestroyedException("Region was destroyed", "pathToRegion");
+    cce.initCause(rde);
+
+    doThrow(cce).when(txState).reserveAndCheck();
+
+    Throwable caughtException = catchThrowable(txState::commit);
+
+    assertThat(caughtException).isInstanceOf(TransactionDataRebalancedException.class);
```

Comment:
This can be condensed to one line, and an additional assertion added to confirm that the thrown exception wraps `cce`:

```suggestion
    assertThatThrownBy(txState::commit).isInstanceOf(TransactionDataRebalancedException.class)
        .hasCause(cce);
```